File: easy/climbing-stairs.ts
// The first submission -> 100ms, 42.7mb
function climbStairs(n: number): number {
    let result = 1;
    let lastResult = 1;
    let tempResult;
    for (let i = 1; i < n; i++) {
        tempResult = result;
        result = result + lastResult;
        lastResult = tempResult;
    }
    return result;
};

/*
1 => 1
2 => 2
3 => 3
4 => 5
5 => 8
6 => 13
7 => 21
8 => 34
9 => 55
==> Fibonacci Sequence
*/

File: easy/climbing-stairs.go
// The third submission -> 0ms, 2.0mb
func climbStairs(n int) int {
    result := 1;
    lastResult := 1;
    tempResult := 0;
    for i := 1; i < n; i++ {
        tempResult = result;
        result = result + lastResult;
        lastResult = tempResult;
    }
    return result;
}